File: prak python/ParkingLot.py
class ParkingLot:
    def __init__(self, capacity):
        self._capacity = capacity
        self._num_vehicles = 0

    @property
    def capacity(self):
        return self._capacity

    @capacity.setter
    def capacity(self, capacity):
        self._capacity = capacity

    @property
    def num_vehicles(self):
        return self._num_vehicles

    @num_vehicles.setter
    def num_vehicles(self, num_vehicles):
        self._num_vehicles = num_vehicles

File: prak cpp/ParkingLot.cpp
#pragma once

#include <iostream>

using namespace std;

class ParkingLot {
private:
    int _capacity;
    int _num_vehicles;

public:
    ParkingLot(int capacity)
        : _capacity(capacity), _num_vehicles(0) {}

    int get_capacity() const {
        return _capacity;
    }

    void set_capacity(int capacity) {
        _capacity = capacity;
    }

    int get_num_vehicles() const {
        return _num_vehicles;
    }

    void set_num_vehicles(int num_vehicles) {
        _num_vehicles = num_vehicles;
    }
};